Snakefile

```diff
@@ -0,0 +1,94 @@
+    This for loop re-implements a simplified veresion of the the CLI of sourmash sig subtract.
+    https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/sig/__main__.py
+    Doing this in a python script is the easiest way to map pairs of samples to each other within the snakemake workflow.
+    
+    The run directive doesn't allow a run to have a conda environment; 
+    therefore the run environment for this code is specified in environment.yml
+    """
+    input:
+        metadata = 'inputs/metadata-paired-mgx-mtx.tsv',
+        sigs = expand('outputs/sourmash_sketch/{run_accession}.sig', run_accession = RUN_ACCESSIONS) 
+    output:
+        sigs = expand("outputs/sourmash_sketch_subtract/{sample}_k{{ksize}}.sig", sample = SAMPLES)
+    run:
+        ksize = wildcards.ksize
+        # read in metadata dataframe to derive sample pairs
+        metadata = pd.read_csv(input.metadata, sep = "\t") # read in metadata as pandas df
+        metadata = metadata.reset_index()  # make sure indexes pair with number of rows
+        
+        for index, row in metadata.iterrows():
+            # grab the run accessions for a given paired metagenome and metatranscriptome
+            mtx_run_accession = row['mtx_run_accession']
+            mgx_run_accession = row['mgx_run_accession']
+            # using the run assession, create paths of the signatures
+            mtx_sigfile = os.path.join('outputs/sourmash_sketch', mtx_run_accession + '.sig')
+            mgx_sigfile = os.path.join('outputs/sourmash_sketch', mgx_run_accession + '.sig')
+
+            # read in mtx signature and grab both the minhash object and the hashes in that object
+            mtx_sigobj = sourmash.load_one_signature(mtx_sigfile, ksize=ksize, select_moltype='DNA')
+            mtx_mh = from_sigobj.minhash
+            # turn the mtx hashes into a set
+            mtx_subtract_mins = set(mtx_mh.hashes)
+
+            # read in mgx signature
+            mgx_sigobj = sourmash.load_one_signature(mgx_sig_path, ksize=ksize, select_moltype='DNA')
+
+            # do the subtraction
+            mtx_subtract_mins -= set(mgx_sigobj.minhash.hashes)
+
+            # make a new signature that only contains the mtx hashes that were not in the mgx
+            mtx_subtract_mh = mtx_sigobj.minhash.copy_and_clear().flatten()
+            mtx_subtract_mh.add_many(mtx_subtract_mins)
+
+            # re-establish hash abundances from mtx sig
+            # re-read in mtx sig just in case abundances were removed in place
+            abund_sig = sourmash.load_one_signature(mtx_sig_path, ksize=ksize, select_moltype='DNA')
+            mtx_subtract_mh = mtx_subtract_mh.inflate(abund_sig.minhash)
+  
+            # create a new sig object that can be written to file
+            mtx_subtract_sigobj = sourmash.SourmashSignature(mtx_subtract_mh)
+
+            # create a name that reflects the signature contents
+            # name = mtx_run_accession + "-minus-" + mgx_run_accession
```

Comment:
Is this supposed to be here?

Reply:
ya. I couldn't decide which name i liked better and I settled on the one that's not commented out but I wasn't ready to totally break up with this one 😂 I should probably remove it

Reply:
```suggestion
```